### PR TITLE
Add a python client library for Bassa

### PR DESCRIPTION
**Pull request by kmehant**
## To do

- [x] User functions
- [x] Download functions
- [ ] File functions
- [x] retries
- [ ] rate limiting

### CODE REVIEW

File: bassa.py [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+                    'API response: {}'.format(result.status_code))
+            elif int(result.status_code/100) == 2:
+                retries = 0
+            elif int(result.status_code/100) == 5:
+                continue
+
+    def get_downloads_user_request(self, limit=1, retries=1):
+        """Get downloads user request
+        Args:
+            limit (int): Number of records to return. limit 1 = 25 records
```
Comment (JaDogg) [inline bassa.py:318]: What does limit 1 = 25 mean?

File: bassa.py [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+    def get_user_signup_requests(self, retries=1):
+        """Get all user requests
+        Args:
+            retires (int): number of retries upon failure
+        Returns:
+            response as json
+        """
+        endpoint = "/api/user/requests"
+        api_url_complete = self.api_url + endpoint
+        while retries != -1:
```
Comment (JaDogg) [inline bassa.py:207]: Can we use some function to handle retry mechanism? You can also try to use a decorator for this.  This seems to be duplicated?

Comment (kmehant) [reply bassa.py:207]: Sure I will use timeouts, retries, request adapters, sessions other advanced features 
Will update it soon

File: bassa.py (hunk 1 above)
Comment (kmehant) [reply bassa.py:318]: It's according to our Bassa server deaign
So if limit is 1 then we can get upto 25 SQL records. 
Max Number of sql records = limit * 25

**Comment by JaDogg** [comment]
I'm merging this @kmehant please do the retry mechanism clean up in a different PR

**Pull request merged**